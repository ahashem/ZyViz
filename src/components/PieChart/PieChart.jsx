import React from 'react';
import PropTypes from 'prop-types';

import { VictoryPie } from 'victory';

/**
 * Abstract Pie Chart wrapper
 * @param data
 */
const PieChart = ({ data, onClick, ...rest }) => {
  return (
    <VictoryPie
      data={data}
      {...rest}
      events={[{
        target: 'data',
        eventHandlers: {
          onClick: (e) => {
            console.log('event', e);
            return [
              {
                target: 'data',
                mutation: (props) => {
                  const fill = props.style && props.style.fill;
                  return fill === '#c43a31' ? null : { style: { fill: '#c43a31' } };
                }
              }, {
                target: 'labels',
                mutation: (props) => {
                  return props.text === 'clicked' ? null : { text: 'clicked' };
                }
              }
            ];
          }
        }
      }]}
    />
  );
};

PieChart.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({})),
  onClick: PropTypes.func,
};

PieChart.defaultProps = {
  data: [],
  onClick: () => {
  },
};

export default PieChart;
