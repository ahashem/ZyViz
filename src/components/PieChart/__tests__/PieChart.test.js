import React from 'react';
import { shallow } from 'enzyme';
import PieChart from '../PieChart';

describe('<PieChart /> component', () => {
  it('should render with simple data', () => {
    const mockChartData = [
      { x: 'Cats', y: 35 },
      { x: 'Dogs', y: 40 },
      { x: 'Birds', y: 55 }
    ];

    const rendered = shallow(
      <PieChart
        name="testChart"
        data={mockChartData}
      />
    );

    expect(rendered).toMatchSnapshot();
  });
});

