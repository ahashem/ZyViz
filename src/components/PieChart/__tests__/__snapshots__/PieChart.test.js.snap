// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart /> component should render with simple data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PieChart
    data={
      Array [
        Object {
          "x": "Cats",
          "y": 35,
        },
        Object {
          "x": "Dogs",
          "y": 40,
        },
        Object {
          "x": "Birds",
          "y": 55,
        },
      ]
    }
    labelKey="key"
    name="testChart"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "colorScale": "cool",
      "containerComponent": <VictoryContainer
        className="VictoryContainer"
        portalComponent={<Portal />}
        portalZIndex={99}
        responsive={true}
      />,
      "data": Array [
        Object {
          "x": "Cats",
          "y": 35,
        },
        Object {
          "x": "Dogs",
          "y": 40,
        },
        Object {
          "x": "Birds",
          "y": 55,
        },
      ],
      "dataComponent": <Slice
        pathComponent={<Path />}
      />,
      "events": Array [
        Object {
          "eventHandlers": Object {
            "onClick": [Function],
          },
          "target": "data",
        },
      ],
      "groupComponent": <g />,
      "labelComponent": <VictoryLabel
        capHeight={0.71}
        direction="inherit"
        lineHeight={1}
        textComponent={<Text />}
        tspanComponent={<TSpan />}
      />,
      "labels": [Function],
      "name": "testChart",
      "padding": 80,
      "sortKey": "key",
      "sortOrder": "descending",
      "standalone": true,
      "theme": Object {
        "area": Object {
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "fill": "#212121",
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "width": 350,
        },
        "axis": Object {
          "height": 350,
          "padding": 50,
          "style": Object {
            "axis": Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            },
            "axisLabel": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
            "grid": Object {
              "fill": "none",
              "pointerEvents": "painted",
              "stroke": "#ECEFF1",
              "strokeDasharray": "10, 5",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
            },
            "tickLabels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "ticks": Object {
              "fill": "transparent",
              "size": 5,
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 1,
            },
          },
          "width": 350,
        },
        "bar": Object {
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "fill": "#455A64",
              "padding": 8,
              "strokeWidth": 0,
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
          },
          "width": 350,
        },
        "boxplot": Object {
          "boxWidth": 20,
          "height": 350,
          "padding": 50,
          "style": Object {
            "max": Object {
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 1,
            },
            "maxLabels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "median": Object {
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 1,
            },
            "medianLabels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "min": Object {
              "padding": 8,
              "stroke": "#455A64",
              "strokeWidth": 1,
            },
            "minLabels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "q1": Object {
              "fill": "#455A64",
              "padding": 8,
            },
            "q1Labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "q3": Object {
              "fill": "#455A64",
              "padding": 8,
            },
            "q3Labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
          },
          "width": 350,
        },
        "candlestick": Object {
          "candleColors": Object {
            "negative": "#455A64",
            "positive": "#ffffff",
          },
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "stroke": "#455A64",
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "width": 350,
        },
        "chart": Object {
          "height": 350,
          "padding": 50,
          "width": 350,
        },
        "errorbar": Object {
          "borderWidth": 8,
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "fill": "transparent",
              "opacity": 1,
              "stroke": "#455A64",
              "strokeWidth": 2,
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "width": 350,
        },
        "group": Object {
          "colorScale": Array [
            "#F4511E",
            "#FFF59D",
            "#DCE775",
            "#8BC34A",
            "#00796B",
            "#006064",
          ],
          "height": 350,
          "padding": 50,
          "width": 350,
        },
        "legend": Object {
          "colorScale": Array [
            "#F4511E",
            "#FFF59D",
            "#DCE775",
            "#8BC34A",
            "#00796B",
            "#006064",
          ],
          "gutter": 10,
          "orientation": "vertical",
          "style": Object {
            "data": Object {
              "type": "circle",
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "title": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 5,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
          },
          "titleOrientation": "top",
        },
        "line": Object {
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "fill": "transparent",
              "opacity": 1,
              "stroke": "#455A64",
              "strokeWidth": 2,
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "width": 350,
        },
        "pie": Object {
          "colorScale": Array [
            "#F4511E",
            "#FFF59D",
            "#DCE775",
            "#8BC34A",
            "#00796B",
            "#006064",
          ],
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "padding": 8,
              "stroke": "#ECEFF1",
              "strokeWidth": 1,
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 20,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
          },
          "width": 350,
        },
        "scatter": Object {
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "fill": "#455A64",
              "opacity": 1,
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 8,
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "width": 350,
        },
        "stack": Object {
          "colorScale": Array [
            "#F4511E",
            "#FFF59D",
            "#DCE775",
            "#8BC34A",
            "#00796B",
            "#006064",
          ],
          "height": 350,
          "padding": 50,
          "width": 350,
        },
        "tooltip": Object {
          "cornerRadius": 5,
          "flyoutStyle": Object {
            "fill": "#f0f0f0",
            "pointerEvents": "none",
            "stroke": "#212121",
            "strokeWidth": 1,
          },
          "pointerLength": 10,
          "style": Object {
            "fill": "#455A64",
            "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
            "fontSize": 12,
            "letterSpacing": "normal",
            "padding": 5,
            "pointerEvents": "none",
            "stroke": "transparent",
            "strokeWidth": 0,
            "textAnchor": "middle",
          },
        },
        "voronoi": Object {
          "height": 350,
          "padding": 50,
          "style": Object {
            "data": Object {
              "fill": "transparent",
              "stroke": "transparent",
              "strokeWidth": 0,
            },
            "flyout": Object {
              "fill": "#f0f0f0",
              "pointerEvents": "none",
              "stroke": "#212121",
              "strokeWidth": 1,
            },
            "labels": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 5,
              "pointerEvents": "none",
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "width": 350,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "colorScale": "cool",
        "containerComponent": <VictoryContainer
          className="VictoryContainer"
          portalComponent={<Portal />}
          portalZIndex={99}
          responsive={true}
        />,
        "data": Array [
          Object {
            "x": "Cats",
            "y": 35,
          },
          Object {
            "x": "Dogs",
            "y": 40,
          },
          Object {
            "x": "Birds",
            "y": 55,
          },
        ],
        "dataComponent": <Slice
          pathComponent={<Path />}
        />,
        "events": Array [
          Object {
            "eventHandlers": Object {
              "onClick": [Function],
            },
            "target": "data",
          },
        ],
        "groupComponent": <g />,
        "labelComponent": <VictoryLabel
          capHeight={0.71}
          direction="inherit"
          lineHeight={1}
          textComponent={<Text />}
          tspanComponent={<TSpan />}
        />,
        "labels": [Function],
        "name": "testChart",
        "padding": 80,
        "sortKey": "key",
        "sortOrder": "descending",
        "standalone": true,
        "theme": Object {
          "area": Object {
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "#212121",
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
            },
            "width": 350,
          },
          "axis": Object {
            "height": 350,
            "padding": 50,
            "style": Object {
              "axis": Object {
                "fill": "transparent",
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 2,
              },
              "axisLabel": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
              "grid": Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              },
              "tickLabels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "ticks": Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              },
            },
            "width": 350,
          },
          "bar": Object {
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "#455A64",
                "padding": 8,
                "strokeWidth": 0,
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
            },
            "width": 350,
          },
          "boxplot": Object {
            "boxWidth": 20,
            "height": 350,
            "padding": 50,
            "style": Object {
              "max": Object {
                "padding": 8,
                "stroke": "#455A64",
                "strokeWidth": 1,
              },
              "maxLabels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "median": Object {
                "padding": 8,
                "stroke": "#455A64",
                "strokeWidth": 1,
              },
              "medianLabels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "min": Object {
                "padding": 8,
                "stroke": "#455A64",
                "strokeWidth": 1,
              },
              "minLabels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "q1": Object {
                "fill": "#455A64",
                "padding": 8,
              },
              "q1Labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "q3": Object {
                "fill": "#455A64",
                "padding": 8,
              },
              "q3Labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
            },
            "width": 350,
          },
          "candlestick": Object {
            "candleColors": Object {
              "negative": "#455A64",
              "positive": "#ffffff",
            },
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "stroke": "#455A64",
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
            },
            "width": 350,
          },
          "chart": Object {
            "height": 350,
            "padding": 50,
            "width": 350,
          },
          "errorbar": Object {
            "borderWidth": 8,
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "transparent",
                "opacity": 1,
                "stroke": "#455A64",
                "strokeWidth": 2,
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
            },
            "width": 350,
          },
          "group": Object {
            "colorScale": Array [
              "#F4511E",
              "#FFF59D",
              "#DCE775",
              "#8BC34A",
              "#00796B",
              "#006064",
            ],
            "height": 350,
            "padding": 50,
            "width": 350,
          },
          "legend": Object {
            "colorScale": Array [
              "#F4511E",
              "#FFF59D",
              "#DCE775",
              "#8BC34A",
              "#00796B",
              "#006064",
            ],
            "gutter": 10,
            "orientation": "vertical",
            "style": Object {
              "data": Object {
                "type": "circle",
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "title": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 5,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
            },
            "titleOrientation": "top",
          },
          "line": Object {
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "transparent",
                "opacity": 1,
                "stroke": "#455A64",
                "strokeWidth": 2,
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
            },
            "width": 350,
          },
          "pie": Object {
            "colorScale": Array [
              "#F4511E",
              "#FFF59D",
              "#DCE775",
              "#8BC34A",
              "#00796B",
              "#006064",
            ],
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "padding": 8,
                "stroke": "#ECEFF1",
                "strokeWidth": 1,
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 20,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
            },
            "width": 350,
          },
          "scatter": Object {
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "#455A64",
                "opacity": 1,
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 8,
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
            },
            "width": 350,
          },
          "stack": Object {
            "colorScale": Array [
              "#F4511E",
              "#FFF59D",
              "#DCE775",
              "#8BC34A",
              "#00796B",
              "#006064",
            ],
            "height": 350,
            "padding": 50,
            "width": 350,
          },
          "tooltip": Object {
            "cornerRadius": 5,
            "flyoutStyle": Object {
              "fill": "#f0f0f0",
              "pointerEvents": "none",
              "stroke": "#212121",
              "strokeWidth": 1,
            },
            "pointerLength": 10,
            "style": Object {
              "fill": "#455A64",
              "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
              "fontSize": 12,
              "letterSpacing": "normal",
              "padding": 5,
              "pointerEvents": "none",
              "stroke": "transparent",
              "strokeWidth": 0,
              "textAnchor": "middle",
            },
          },
          "voronoi": Object {
            "height": 350,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "transparent",
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "flyout": Object {
                "fill": "#f0f0f0",
                "pointerEvents": "none",
                "stroke": "#212121",
                "strokeWidth": 1,
              },
              "labels": Object {
                "fill": "#455A64",
                "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                "fontSize": 12,
                "letterSpacing": "normal",
                "padding": 5,
                "pointerEvents": "none",
                "stroke": "transparent",
                "strokeWidth": 0,
                "textAnchor": "middle",
              },
            },
            "width": 350,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
